test/integration/targets/ec2_eip/tasks/main.yml

```diff
@@ -35,15 +35,15 @@
         that:
           - new_eip is defined
           - new_eip is changed
-          - new_eip.public_ip is defined and new_eip.public_ip != ""
-          - new_eip.allocation_id is defined and new_eip.allocation_id != ""
+          - new_eip.public_ip is defined and new_eip.public_ip is match("^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$")
```

Comment:
Thinking about it, this might be better as
```
- new_eip.public_ip is defined and (new_eip.public_ip | ipaddr)
```

Reply:
Clever! I've updated the tests accordingly :+1: